presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -0,0 +1,156 @@
+        implements ConnectorSplitManager
+{
+    static final long DAYS_IN_A_YEAR = 365;
+    static final long DAYS_IN_A_WEEK = 7;
+    private final PrometheusClient prometheusClient;
+
+    @Inject
+    public PrometheusSplitManager(PrometheusClient prometheusClient)
+    {
+        this.prometheusClient = requireNonNull(prometheusClient, "client is null");
+        requireNonNull(prometheusClient.config.getQueryChunkSizeDuration(), "query-chunk-size-duration is null");
+        requireNonNull(prometheusClient.config.getMaxQueryRangeDuration(), "max-query-range-duration is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorTableHandle connectorTableHandle,
+            SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        PrometheusTableHandle tableHandle = (PrometheusTableHandle) connectorTableHandle;
+        PrometheusTable table = prometheusClient.getTable(tableHandle.getSchemaName(), tableHandle.getTableName());
+
+        // this can happen if table is removed during a query
+        if (table == null) {
+            throw new TableNotFoundException(tableHandle.toSchemaTableName());
+        }
+        List<ConnectorSplit> splits = new ArrayList<>();
+        generateTimesForSplits(Instant.now().toEpochMilli()).forEach(time -> {
+            try {
+                splits.add(new PrometheusSplit(buildQuery(prometheusClient.config.getPrometheusURI(), time, table.getName())));
+            }
+            catch (URISyntaxException e) {
+                throw new UnsupportedOperationException(e.getMessage());
+            }
+        });
+        return new FixedSplitSource(splits);
+    }
+
+    // URIBuilder handles URI encode
+    private URI buildQuery(URI baseURI, String time, String tableName)
+            throws URISyntaxException
+    {
+        List<NameValuePair> nameValuePairs = new ArrayList<>(2);
+        nameValuePairs.add(new BasicNameValuePair("query",
+                tableName + "[" + prometheusClient.config.getQueryChunkSizeDuration() + "]"));
+        nameValuePairs.add(new BasicNameValuePair("time",
+                time));
+        return new URIBuilder(baseURI.toString())
+                .setPath("api/v1/query")
+                .setParameters(nameValuePairs).build();
+    }
+
+    /**
+     * Get the start times in seconds that divide up the query using the config value:
+     * maxQueryRangeDuration
+     * queryChunkSizeDuration
+     * The times will be used in queries to Prometheus like: `http://localhost:9090/api/v1/query?query=up[3w]&time=1568229904"`
+     * ** NOTE: Prometheus instant query wants the end time specified, you get duration before that time returned
+     *
+     * @return list of start times as epoch seconds, like ["1568053244", "1568926595"]
+     */
+    private List<String> generateTimesForSplits(long nowTimeInMillis)
```

Comment:
I'd like to see this unit tested. I think if we made it independent of the PrometheusSplitManager class that would be easier since all the config options can be made into arguments.

Reply:
Unit tests added in 029f46a